### DISCUSSION TITLE
eslint-plugin-vitest

### DISCUSSION THREAD

**Discussion by Boshen** [oxc-project/oxc#3359] [Feature Request]
https://www.npmjs.com/package/eslint-plugin-vitest

We need to somehow make jest plugin work with vitest cc @mysteryven if you plan to refactor some of the code

**Comment by mysteryven** [comment]
~~I am planning to refactor [`parseJestFnCall`](https://github.com/oxc-project/oxc/blob/f6f7adc07fa3ddc3dc2aa9c0d123ef16f424ad15/crates/oxc_linter/src/utils/jest/parse_jest_fn.rs#L17), thanks for letting me pay attention to this.~~

As take a rough look on vitest recommend rules, [`no-import-node-test`](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/no-import-node-test.md), [`require-local-test-context-for-concurrent-snapshots`](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/require-local-test-context-for-concurrent-snapshots.md) are additional, [`no-commented-out-tests`](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/no-commented-out-tests.md),  [`valid-describe-callback`](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/valid-describe-callback.md), [`no-alias-methods`](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-alias-methods.md) are almost equal, the remain 3 rules need slightly change. Even though the implementation is very similar, the test cases are different.

~~Copy them will let many codes repeat twice, I am thinking about how to organize them, not have good idea now.~~

**Comment by Sysix** [comment]
closing the discussion because we are already supporting it: https://github.com/oxc-project/oxc/issues/4656
